src/graphql/resolver.ts

```diff
@@ -0,0 +1,92 @@
+import * as queries from "../db/db-queries";
+import * as jwt from "jsonwebtoken";
+import * as UserCart from '../socket/cart';
+import {getUpdatedProductsAmount} from '../socket/cart';
+import {getUpdatedProductAmount} from "../socket/socket-controllers";
+import {Product} from "../models/product";
+import {PubSub} from 'apollo-server-koa';
+
+const pubsub = new PubSub();
+
+export const updateProduct = (parent, {product}, context) => {
+    return pubsub.asyncIterator(['PRODUCT_UPDATED']);
+};
+export const resolvers = {
+
+    Query: {
+        getProducts: (parent, args, context, info) => {
+            if (context.token) {
+                return queries.findProductsQuery().then(dbProducts => {
+                    return getUpdatedProductsAmount(dbProducts);
+                });
+            }
+            return null;
+        },
+        getProduct: (parent, {_id}, context, info) => context.token ? queries.findProductByIdQuery(_id) : null
+    },
+    Mutation: {
+        addProduct: (parent, {product}, context, info) => {
+            if (context.token) {
```

Comment:
first throw exception if token not found. second think about other way to handle unauthorized mutations or queries